### DISCUSSION TITLE
Build when method

### DISCUSSION THREAD

**Discussion by MaldonadoAndres** [rrousselGit/riverpod#1726] [Q&A] [answered]
Is there any way to tell my build method to only do a rebuild when certain states are issued?

**Comment by rrousselGit** [comment, marked as answer]
You can use `ref.watch(provider.select((value) => value.someProperty))` to only rebuild when `someProperty` changes

**Reply by MaldonadoAndres** [reply]
I'm doing this for testing purposes and it only rebuilds the consumer when I manually hot reload the app 
![Screen Shot 2022-10-04 at 1 48 27 PM](https://user-images.githubusercontent.com/42980462/193901322-efd981dc-0059-46f7-ad75-3cd71247a9ae.png)
I need to user provider.notifier because if I only use the provider the value it's just the states of the provider.
PD: I'm using a StateNotifier

**Reply by rrousselGit** [reply]
You're misusing StateNotifierProvider. You should be using provider.notifier here, that's your problem.

**Reply by MaldonadoAndres** [reply]
Can you elaborate a little more on that?
By the way, is this `final accounts = ref.watch(ledgerProvider.notifier).accounts` what you mean by "You should be using provider.notifier here" because it just behave the same as before

**Reply by rrousselGit** [reply]
Sorry I meant shouldn't

Don't use ".notifier" here.

**Reply by MaldonadoAndres** [reply]
In that case, as you can see the value is just the API for the states generated using freezed and using whenOrNull when the states changes it return null again to my variable
![Screen Shot 2022-10-04 at 2 28 54 PM](https://user-images.githubusercontent.com/42980462/193908639-5dcdecd1-5c7a-4f8a-9d7b-3e16c972faf9.png)

**Reply by rrousselGit** [reply]
You can use "whenData" to transform the AsyncValue and have it contain only what you need

**Reply by MaldonadoAndres** [reply]
Can you give an example or redirect me to the page when I can read about it I can't find anything about that in the documentation

**Reply by MaldonadoAndres** [reply]
Nvm I find an answer on stackoverflow that explain how to do it and it works perfectly.
Thanks a lot for your help with this you  literally just save my job 🤣

**Reply by cbenhagen** [reply]
@MaldonadoAndres would you mind sharing your solution for others who stumble upon your question?

**Reply by dleurs** [reply]
Possible to also get the previous state value inside .select((value) ?
Because we also want to rebuild only if the previous state argument is different that the current state argument

**Reply by gdurandrexel** [reply]
Isn't it the behavior of select() already? It checks the equality of the values returned by the selector. Of course you need a proper implementation of `==` for the returned value. And note that you are not forced to return a field of your state: it can be any object.

**Reply by dleurs** [reply]
No for example :
0 : ref.watch(provider.select((state) => state.variableA == true)
1. MyState(variableA: false, variableB: false)  -> No rebuild
2. MyState(variableA: true, variableB: false)  -> Rebuild
3. MyState(variableA: true, variableB: true)  -> Rebuild again but it was not necessary

I would like ref.watch(provider.select((previous, current) => previous.variableA != current.variableA && current.variableA == true)
